### PR TITLE
feat(rust): add c binding

### PR DESCRIPTION
**Pull request by SanjoDeundiak**

### CODE REVIEW

File: implementations/rust/c/bindings/src/ctypes.rs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#![allow(non_camel_case_types)]
```
Comment (mikelodder7) [inline implementations/rust/c/bindings/src/ctypes.rs:1]: Not sure if this is necessary with [this](https://doc.rust-lang.org/std/os/raw/index.html).
Can we use the built-in rust ones instead?

File: implementations/rust/c/rust_memory/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    ERROR_NONE
+}
+
+unsafe extern "C" fn memcpy_impl(
+    _: *mut ockam_memory_t,
+    dst: *mut core::ffi::c_void,
+    src: *const core::ffi::c_void,
+    size: usize,
+) -> ockam_error_t {
+    core::intrinsics::copy_nonoverlapping(src, dst, size);
```
Comment (mikelodder7) [inline implementations/rust/c/rust_memory/src/lib.rs:112]: Is there a reason we're using `core::intrinsics` vs [std::ptr](https://doc.rust-lang.org/beta/std/ptr/index.html)?

`core::instrinsics` only works on nightly and I believe we are targeting stable

File: implementations/rust/c/rust_memory/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+    ERROR_NONE
+}
+
+unsafe extern "C" fn memmove_impl(
+    _: *mut ockam_memory_t,
+    dst: *mut core::ffi::c_void,
+    src: *mut core::ffi::c_void,
+    size: usize,
+) -> ockam_error_t {
+    core::intrinsics::copy(src, dst, size);
```
Comment (mikelodder7) [inline implementations/rust/c/rust_memory/src/lib.rs:122]: Same as before.

**Review by mikelodder7 — COMMENTED** [review]
Great PR! Just a few questions to answer then I think it can be merged.

File: implementations/rust/c/bindings/src/ctypes.rs (hunk 1 above)
Comment (SanjoDeundiak) [reply implementations/rust/c/bindings/src/ctypes.rs]: Yes, fixed

File: implementations/rust/c/rust_memory/src/lib.rs (hunk 2 above)
Comment (SanjoDeundiak) [reply implementations/rust/c/rust_memory/src/lib.rs]: Changed to std::ptr

File: implementations/rust/c/rust_memory/src/lib.rs (hunk 3 above)
Comment (SanjoDeundiak) [reply implementations/rust/c/rust_memory/src/lib.rs]: +

**Pull request merged**